src/util/orm.py

```diff
@@ -289,14 +289,16 @@ def recompute_fields(cr, model, fields, ids=None, logger=_logger, chunk_size=256
 
     if ids is None:
         query = format_query(cr, "SELECT id FROM {} ORDER BY id", table_of_model(cr, model)) if query is None else query
-        cr.execute(query)
-        count = cr.rowcount
-        if count < 2**21:  # avoid the overhead of a named cursor unless we have at least two chunks
+        count_query = "SELECT count(*) FROM ({query})".format(query=query)
+        cr.execute(count_query)
+        count = cr.fetchone()[0]
+        itersize = 100000
```

Comment:
Maybe we can reuse BIG_TABLE_THRESHOLD
```suggestion
        itersize = BIG_TABLE_THRESHOLD
```
Or a factor of it
```suggestion
        itersize = 2 * BIG_TABLE_THRESHOLD
```